rfc/draft-oberstet-hybi-tavendo-wamp-00.md

```diff
@@ -41,91 +41,84 @@ This document defines the Web Application Messaging Protocol (WAMP). WAMP is a r
 
-Combining these two patterns into a single protocol allows a single protocol to be used for the entire messaging requirements of an application, reducing complexity in the technology stack and networking overheads.
+Combining these two patterns into a single protocol allows it to be used for the entire messaging requirements of an application, thus reducing technology stack complexity, as well as networking overheads.
 
-While WAMP was originally specified to run over WebSocket, it can run over any transport which is message-based, ordered, reliable and bi-directional.
+While WAMP was originally specified to run over WebSocket, it can run over any transport which is message-oriented, ordered, reliable, and bi-directional.
 
 ## Protocol Overview
 
 _This section is non-normative._
 
-For each of the two messaging patterns, three roles are defined:
+The PubSub messaging pattern defines three roles: _Subscribers_ and _Publishers_, which communicate via a _Broker_.
 
-For PubSub, there are Subscribers and Publishers, which are connected through a Broker.
+The routed RPC messaging pattern also defines three roles: _Callers_ and _Callees_, which communicate via a _Dealer_.
 
-For routed Remote Procedure Calls there are Callers and Callees, which are connected through a Dealer.
+WAMP Connections are established by Clients to a Router. Connections can use any transport that is message-based, ordered, reliable and bi-directional, with WebSocket as the default transport.
 
-WAMP Connections are established by Clients to a Router. Connections can use any transport which is message-based, ordered, reliable and bi-directional, with WebSocket as the default transport.
+A _Router_ is a component which implements one or both of the Broker and Dealer roles. A _Client_ is a component which implements any or all of the Subscriber, Publisher, Caller, or Callee roles.
 
-WAMP Sessions are established using a WAMP Connection. A WAMP Session connects to a Realm on a Router. Routing occurs only between WAMP Sessions connected to the same realm.
++WAMP _Connections_ are established by Clients to a Router. Connections can use any transport which is message-oriented, ordered, reliable and bi-directional, with WebSocket as the default transport.
 
-The Basic Profile defines the parts of the protocol which are required to establish a WAMP connection as well as for basic interactions between the four client and two router roles. 
++WAMP _Sessions_ are established over a WAMP Connection. A WAMP Session is joined to a _Realm_ on a Router. Routing occurs only between WAMP Sessions that have joined the same Realm.
```

Comment:
I think we should use the term "connect" within the context of transports, and the term "join" within the context of realms.